Lib/_py_abc.py

```diff
@@ -134,14 +134,20 @@ def __subclasscheck__(cls, subclass):
             return True
         # Check if it's a subclass of a registered class (recursive)
         for rcls in cls._abc_registry:
-            if issubclass(subclass, rcls):
-                cls._abc_cache.add(subclass)
-                return True
+            try:
+                if issubclass(subclass, rcls):
+                    cls._abc_cache.add(subclass)
+                    return True
+            except Exception:
```

Comment:
In the C code I actually suppress all exceptions, not just Exception. Not sure what the right thing to do is.

Reply:
I'd say suppressing all exceptions would definitely be bad -- that would mean that we'd be suppressing e.g. `KeyboardInterrupt`, and `SystemExit`.

Honestly, I feel pretty uncomfortable about suppressing all subclasses of `Exception` -- could we just do `TypeError`? If I had a typo in a custom `__subclasscheck__` method, I'm not sure I'd want e.g. the resulting `AttributeError` to be silenced because of the fact that `ABCMeta` was being used somewhere